src/Chirp.Web/Pages/TimelineModel.cshtml.cs

```diff
@@ -1,31 +1,37 @@
-﻿using Microsoft.AspNetCore.Authorization;
-using Microsoft.AspNetCore.Mvc;
-using Microsoft.AspNetCore.Mvc.RazorPages;
-using Chirp.Core;
-using Chirp.Infrastructure;
-
 namespace Chirp.Web.Pages;
 
-public class UserTimelineModel : PageModel
+public class TimelineModel : PageModel
 {
-    private readonly ICheepService _service;
     public List<Cheep> Cheeps { get; set; } = new List<Cheep>();
     public int CheepsPerPage;
     public int NumOfCheeps;
-    private ChirpDBContext _db;
-    private readonly ICheepRepository _cheepRepository;
-    private readonly IAuthorRepository _authorRepository;
-    public UserTimelineModel(ChirpDBContext db, ICheepRepository cheepRepository, IAuthorRepository authorRepository, ICheepService service)
+    protected readonly ICheepService _service;
+    protected ChirpDBContext _db;
+    protected readonly ICheepRepository _cheepRepository;
+    protected readonly IAuthorRepository _authorRepository;
+    public TimelineModel(ChirpDBContext db, ICheepRepository cheepRepository, IAuthorRepository authorRepository, ICheepService service)
     {
         _db = db;
         _cheepRepository = cheepRepository;
         _authorRepository = authorRepository;
         _service = service;
     }
 
-    public ActionResult OnGet(string author, [FromQuery] int page = 1)
+    public async Task<IActionResult> OnPostAsync()
     {
+        string text = Request.Form["Text"];
+        if (text.Length > 180) text = text.Substring(0, 180);
 
+        _authorRepository.CreateAuthor(User.Identity.Name, "example@mail.com");
```

Comment:
Why are we creating an author each time we post?

Reply:
The CreateAuthor method creates the user as an Author in case it doesn't already exist in the database. Ideally, this should only happen when a new user logs in, but I don't know how to do that.